### PR TITLE
Ignore enumerators that return no monitors

### LINKED ISSUE #58: get_monitors returns empty on osx when Enumerator not explicitly provided
tl;dr:
```py
>>> from screeninfo import get_monitors, Enumerator
>>> get_monitors()
[]
>>> get_monitors(Enumerator.OSX)
[Monitor(x=0, y=0, width=1680, height=1050, width_mm=None, height_mm=None, name=None)]
```

On those lines:
https://github.com/rr-/screeninfo/blob/master/screeninfo/screeninfo.py#L29-L33
```py
    for enumerator in Enumerator:
        try:
            return _get_monitors(enumerator)
        except Exception:
            pass
```

Seems some of the previous enumerator is not raising, so it returns before trying OSX, one solution is to also check for empty return on that try/catch

### PR DESCRIPTION
**Pull request by rr-**
Fixes #58

### CODE REVIEW

File: screeninfo/screeninfo.py
Comment (betafcc) [inline screeninfo/screeninfo.py:23]: Possible refactor:

```suggestion
    if name is not None:
        return list(ENUMERATOR_MAP[Enumerator(name)].enumerate_monitors())
```

**Review by betafcc — APPROVED** [review]
Works as expected os macOS and tests passes, LGTM 🚀

File: screeninfo/screeninfo.py
Comment (rr-) [reply screeninfo/screeninfo.py]: Thanks. Fixed. I'll hold off with merging this until we land #60

**Pull request merged**
